ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/SqlFunctionConverter.java

```diff
@@ -349,7 +351,7 @@ private static String getName(GenericUDF hiveUDF) {
   /**
    * This class is used to build immutable hashmaps in the static block above.
    */
-  private static class StaticBlockBuilder {
+  private static class StaticBlockBuilder implements Consumer<Pair<String, SqlOperator>> {
```

Comment:
Is the intention to move all function to use this mechanism?

I am also a bit confused since this block does not seem to contain Aggregate functions. Will we be using same approach for those other aggregate functions?

Reply:
I didn't wanted to pollute this class with the registering of that stuff - because registering all agreggates(which are mergeable) for all sketches is a bit more than just registering a single function

So the registering of the aggregates happens thru the `Consumer.accept()` invocation

Reply:
I've moved to a plugin based mechanism; see the latest patch...